packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx

```diff
@@ -122,29 +124,17 @@ const Attachment: React.FC<WithAsProp<AttachmentProps>> &
 
   const element = (
     <ElementType {...getA11Props('root', { className: classes.root, onClick: handleClick, ...unhandledProps })}>
-      {Box.create(icon, {
-        defaultProps: () => ({ styles: resolvedStyles.icon }),
-      })}
+      {AttachmentIcon.create(icon)}
+
       {(header || description) && (
-        <div className={classes.content}>
-          {Text.create(header, {
-            defaultProps: () => ({ styles: resolvedStyles.header }),
-          })}
-
-          {Text.create(description, {
-            defaultProps: () => ({ styles: resolvedStyles.description }),
-          })}
+        <div className="ui-attachment__content">
+          {AttachmentHeader.create(header)}
+          {AttachmentDescription.create(description)}
         </div>
       )}
-      {Button.create(action, {
-        defaultProps: () => ({
-          iconOnly: true,
-          text: true,
-          styles: resolvedStyles.action,
-          className: Attachment.slotClassNames.action,
-        }),
-      })}
-      {!_.isNil(progress) && <div className={classes.progress} style={{ width: `${progress}%` }} />}
+
+      {AttachmentAction.create(action)}
```

Comment:
It's interesting that none of the sub-components need any of the parent's props :)